File: 1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int):
        buff_dict = {}
        for i in range(len(nums)):
            if nums[i] in buff_dict:
                return [buff_dict[nums[i]], i]
            buff_dict[target - nums[i]] = i

File: 1/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] holder = new int[2048];
        for (int i = 0; i < nums.length; i++) {
            int delta = (target - nums[i]) & 2047;
            if (holder[delta] != 0) {
                return new int[]{holder[delta]-1, i};
            } else {
                holder[nums[i] & 2047] = i+1; 
            }
        }
        return new int[]{-1,-1};
    }
}